src/util/CartesianUtils.ts

```diff
@@ -32,15 +33,47 @@ export const formatAxisMap = (props: any, axisMap: any, offset: any, axisType: A
     const { orientation, domain, padding = {}, mirror, reversed } = axis;
     const offsetKey = `${orientation}${mirror ? 'Mirror' : ''}`;
 
-    let range, x, y, needSpace;
+    let calculatedPadding, range, x, y, needSpace;
+
+    if (axis.type === 'number' && (axis.padding === 'gap' || axis.padding === 'no-gap')) {
+      const diff = domain[1] - domain[0];
+      let smallestDistanceBetweenValues = Infinity;
```

Comment:
`axis.categoricalDomain.length === 1` can be a special case here.

Reply:
Can you elaborate? In case of `axis.categoricalDomain.length === 1` the `smallestDistanceBetweenValues` will be the value of the single element